ui/components/settings/SettingsForm.vue

```diff
@@ -242,16 +243,37 @@ export default {
       "beta_2_mode",
       "user_account",
       "stored_customer_id",
-      "pulse3d_versions",
-      "pulse3d_version_selection_index",
+      "pulse3d_version_info",
+      "selected_pulse3d_version",
       "job_limit_reached",
       "stored_usernames",
       "auto_upload",
       "auto_delete",
       "run_recording_snapshot_default",
     ]),
     sorted_pulse3d_versions: function () {
-      return semver_sort.desc(this.pulse3d_versions);
+      const mapped_versions = {};
+      this.pulse3d_version_info.map((info) => {
+        mapped_versions[info.version] = info;
+      });
+      const external_versions = semver_sort.desc(
+        this.pulse3d_version_info.filter(({ state }) => state === "external").map(({ version }) => version)
+      );
+      const testing_versions = semver_sort.desc(
+        this.pulse3d_version_info.filter(({ state }) => state === "testing").map(({ version }) => version)
+      );
+      const deprecated_versions = semver_sort.desc(
+        this.pulse3d_version_info.filter(({ state }) => state === "deprecated").map(({ version }) => version)
```

Comment:
because we're doing these `.map`s here? (2/2)

Reply:
Yes, we need to store all the information that comes back from that response and filter later